### ISSUE TITLE
Do not add logging handler when there is already a default handler.

### ISSUE THREAD

**Issue by anntzer** [Theano/Theano#1815]
Theano currently always attaches a handler to the "theano" (base) logger.  However, it should first check if another handler is already present (for Python 3.2+, one can use `getLogger().hasHandlers`; otherwise the code can be backported, see e.g. ipython/ipython#2917), as applications that make use of theano may want to control logging output themselves.

**Comment by knighton** [comment]
+1 - libraries should not not as a practice configure logging directly, otherwise they clobber pre-initialized loggers setup by the application developer.

**Comment by nouiz** [comment]
I don't think we will have the time shortly to do this. Do you want to help?

**Comment by nouiz** [comment]
We won't have time to work on this, but if someone have time for this, it would be great to have this done.

**Comment by joaovictortr** [comment]
I've started working on this issue. This is my first attempt at contributing with the project.

So far I've managed to remove the handler when another one is detected (as suggested above). However, since the handler is checked when the theano module is imported (when logging.basicConfig has not yet been called), the above example didn't work. In order to provide the ability to manually disable the handler, I've also added a disable_log_handler function that removes theano_logger's default handler.

So the above example becomes: 
```python
import logging, theano

logging.basicConfig(format="%(asctime)s:%(levelname)s:%(name)s: %(message)s")

theano.disable_log_handler()

theano.theano_logger.warning("foo") # Usually it's a library function that does this.
```
Which produces:
```
2017-02-14 21:42:04,020:WARNING:theano: foo
```

The pull request is available at #5548

I'm not sure if this is the proper way to do it and any feedback/guidance is very appreciated.
